### DISCUSSION TITLE
Avoid re-indexing on restart

### DISCUSSION THREAD

**Discussion by nipunarora** [photoprism/photoprism#1941] [Q&A] [answered]
I just restarted my docker container without moving any of the photos, and it seems like I need to re-index. Is this something that can be avoided?

**Comment by lastzero** [comment, marked as answer]
Check your configuration. Did you mount / define a permanent storage folder for PhotoPrism and the database?

**Reply by nipunarora** [reply]
Thanks @lastzero - it actually does seem to not re-index on restarts with the same folder location.

**Comment by nipunarora** [comment]
I basically moved the folder and `docker-compose.yml` into a subfolder and re-started from there. But the storage folder location seems to be relative to the docker pwd, so it should work the same?

**Reply by graciousgrey** [reply]
You can define the location of the storage folder in the docker-compose.yml within the volumes section

**Comment by nipunarora** [comment]
quick follow up question if I add new original folders, but keep the existing ones the same, should I expect existing folders to be indexed again?

**Reply by graciousgrey** [reply]
A complete rescan will index all originals. A normal index will index only new or edited files (https://docs.photoprism.app/user-guide/library/originals/#when-should-complete-rescan-be-selected)
